### DISCUSSION TITLE
Any way to offer File Download links or pop up generated plots?

### DISCUSSION THREAD

**Discussion by cycleuser** [zauberzeug/nicegui#177] [Q&A]
```Python

diagram = ui.plot()
x = np.linspace(0.0, 5.0)
y = np.cos(2 * np.pi * x) * np.exp(-x)
plt.plot(x, y, '-')
plt.xlabel('time (s)')
plt.ylabel('Damped oscillation')
diagram.view.set_figure(plt.gcf())

plt.savefig('vector.svg')
ui.link('Vector File', 'vector.svg')
```

Now the ui.link can not offer a valid link for user to download generated plot as svg( or any other vector image)>.<

**Comment by falkoschindler** [comment]
You have to *serve* the file (or folder) to make it accessible from a client. The easiest way is to use `ui.add_static_files`:

https://nicegui.io/reference#static_files

Note that the link should include the given route, e.g. "/static".

**Reply by cycleuser** [reply]
static files will not work when the svg generated by uploaded file.
remember my last post on getting the upload path?
that is the whole story: upload a datasheet, generate a plot, offer to user.

**Reply by falkoschindler** [reply]
But the directory where all generated files are stored is known beforehand. So the static route can be created, even though the files will be created later.

**Reply by cycleuser** [reply]
I found out the key now.
It is related to the location where using python to run the ui.run() instead of the location of the py files.
It seems no way to set as a path relatively fixed to the source code py file.
